app/src/main/java/com/github/se/cyrcle/model/parking/ParkingRepositoryFirestore.kt

```diff
@@ -120,11 +121,53 @@ class ParkingRepositoryFirestore @Inject constructor(private val db: FirebaseFir
       onSuccess: () -> Unit,
       onFailure: (Exception) -> Unit
   ) {
-    db.collection(collectionPath)
-        .document(id)
-        .delete()
-        .addOnSuccessListener { onSuccess() }
-        .addOnFailureListener { onFailure(it) }
+    val parkingDocRef = db.collection(collectionPath).document(id)
+    val reportsCollectionRef = parkingDocRef.collection("reports")
+
+    // Recursive function to delete all documents in a collection
+    fun deleteSubcollection(
+        collectionRef: CollectionReference,
+        onComplete: () -> Unit,
+        onError: (Exception) -> Unit
+    ) {
+      collectionRef
+          .get()
+          .addOnSuccessListener { querySnapshot ->
+            val batch = db.batch()
+
+            for (document in querySnapshot.documents) {
+              batch.delete(document.reference)
+            }
+
+            batch
+                .commit()
+                .addOnSuccessListener {
+                  if (querySnapshot.size() < 500) {
```

Comment:
what's happening here ?

Reply:
This is the condition that will allow us to verify that the subcollection is succesfully cleared: once the "reports" are indeed cleared, we can safely delete the parent collection